internal/metrics/parser.go

```diff
@@ -0,0 +1,54 @@
+// Copyright © 2020 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+// ParseOpenConnections returns the sum of open connections from a Prometheus HTTP request
```

Comment:
The name `ParseOpenConnections`, which is very specific doesn't really match the parameterization, which seems to indicate this is a general function. I'd be inclined to remove the second two arguments and hard-code them as constants. That would make the mapping from there to the envoy metrics clearer, and we wouldn't really lose much flexibility.